### PR TITLE
fakeBattery: Add RPC port to change the values returned

### PR DESCRIPTION
**Pull request by drdanz**
#### `fakeBattery`

* Added a new port `<name>/control/rpc:i` to change the values reported by the
  fake battery.
* The battery is now charged/discharged depending on the sign of the current.

### CODE REVIEW

**Comment by drdanz** [comment]
CC @miccol

**Comment by drdanz** [comment]
Depends on #2212

File: src/devices/fakeBattery/fakeBattery.cpp [hunk 1]
```diff
@@ -134,7 +116,60 @@ bool FakeBattery::getBatteryInfo(string& info)
...
+}
+
+void FakeBattery::setBatteryTemperature(const double temperature)
 {
-    yTrace("FakeBattery Thread released\n");
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_temperature = temperature;
+}
+
+void FakeBattery::updateStatus()
```
Comment (randaz81) [inline src/devices/fakeBattery/fakeBattery.cpp]: I wonder if this whole method could be put in a base class, so other battery devices (e.g. bcbBattery, bmsBattery) can reuse it. (Maybe in the next PR....)

File: src/devices/fakeBattery/fakeBattery.cpp [hunk 2]
```diff
@@ -134,7 +116,60 @@ bool FakeBattery::getBatteryInfo(string& info)
...
+            battery_status = yarp::dev::IBattery::Battery_status::BATTERY_OK_IN_USE;
+        } else if (battery_charge > 5.0) {
+            battery_status = yarp::dev::IBattery::Battery_status::BATTERY_LOW_WARNING;
+        } else {
+            battery_status = yarp::dev::IBattery::Battery_status::BATTERY_CRITICAL_WARNING;
+        }
+    } else if (battery_current < -0.1) {
+        battery_status = yarp::dev::IBattery::Battery_status::BATTERY_OK_STANBY;
+    } else {
+        battery_status = yarp::dev::IBattery::Battery_status::BATTERY_OK_IN_CHARGE;
```
[the remark this answers is not in the document]
Comment (drdanz) [reply src/devices/fakeBattery/fakeBattery.cpp]: :+1: 
Thanks for catching this...
Apparently the status is not used by `yarpbatterygui`, which makes it quite difficult to debug the status

**Pull request merged**